enhancements/storage/csi-efs-operator.md

```diff
@@ -0,0 +1,339 @@
+build and shipment), and from user perspective.
+
+## Motivation
+
+Some of our customers want to use Amazon Elastic FileSystem in their clusters. At the same time, not all customers want
+to use it, so it should not be installed by default as our AWS EBS CSI driver.
+
+Despite written by Red Hat employees, the existing CSI driver operator
+[aws-efs-operator](https://github.com/openshift/aws-efs-operator) uses upstream CSI driver, upstream CSI sidecars and
+[the operator itself](https://quay.io/repository/app-sre/aws-efs-operator?tab=tags) is not released through
+registry.redhat.com and therefore cannot be supported by us.
+
+### Goals
+
+* Allow users to use AWS EFS volumes in a supported way.
+* Allow existing users of community-supported aws-efs-operator to update to Red Hat supported solution.
+* Keep existing community community-supported aws-efs-operator features (namely, `SharedVolume` in
+  `aws-efs.managed.openshift.io` API group). See below.
+
+### Non-Goals
+
+* Change the EFS CSI driver in any way. All missing functionality there must go through RFE process.
+
+## Existing situation
+
+### [aws-efs-operator](https://github.com/openshift/aws-efs-operator)
+
+Summary of current [aws-efs-operator](https://github.com/openshift/aws-efs-operator) (no judging, just stating the facts):
+
+* It's written using `controller-runtime`.
+* Once the operator is installed by OLM, the operator automatically installs the CSI driver, without any CRD/CR.
+  * As consequence, user cannot un-install the CSI driver easily.
+* By default, it is installed in `openshift-operators` namespace and the CSI driver runs there too.
+* It does not create EFS volumes in AWS! It's up to the cluster admin to create EFS volume in AWS, figure in which VPC
+  it should be available and set correct security group. See [this KB article](https://access.redhat.com/articles/5025181)
+  for suggested procedure.
+* It offers `SharedVolume` CRD in `aws-efs.managed.openshift.io/v1alpha1` API group to create PV + PVC for an EFS
+  volume. The EFS volume still needs to be created manually, see the previous bullet!
+  * All that `SharedVolume` really does is that it creates PV with proper settings + a PVC. This allows unprivileged
+    users to create PVs. Cluster admins need to be careful to give permissions to use this CRD only to trusted users.
+  * Using `SharedVolume` is purely optional! Cluster admin may create PVs for EFS volumes manually, it's not harder
+    than using `SharedVolume`.
+
+### AWS EFS CSI driver
+
+AWS EFS CSI driver is different from the other CSI drivers we ship. Listing the main differences here for completeness.
+
+* It implements dynamic provisioning in non-standard way. Cluster admin must manually create an EFS share in the cloud
+  (and figure out related networking and firewalls). The EFS CSI driver will provision volumes out of it as
+  *subdirectories* of it. The CSI driver cannot create EFS shares on its own.
```

Comment:
This is no longer true. The upstream driver [supports dynamic provisioning](https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/examples/kubernetes/dynamic_provisioning) as of version 1.2.0. The implementation looks a little weird to me, so you might consider rethinking this design with that in mind.

Reply:
The implementation really creates subdirs in an existing EFS share. The cluster admin still must create an initial share first.